GCC 15: error: address of automatic variable 'transaction' can escape to 'musttail' call

With GCC 15, the following errors are returned when building the current master branch (also seen in v1.18.2):

```
  CC       src/libdqlite_la-leader.lo
src/leader.c: In function 'exec_tick':
src/leader.c:512:37: error: address of automatic variable 'transaction' can escape to 'musttail' call [-Werror=maybe-musttail-local-addr]
  512 |                         TAIL return req->work_cb(req);
      |                                     ^~~~~~~~~~~~~~~~~
src/leader.c:583:45: error: address of automatic variable 'transaction' can escape to 'musttail' call [-Werror=maybe-musttail-local-addr]
  583 |                                 TAIL return exec_tick(req);
      |                                             ^~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:3270: src/libdqlite_la-leader.lo] Error 1
```

```
$ gcc --version
gcc (Debian 15.2.0-3) 15.2.0
```

--- a/src/leader.c
+++ b/src/leader.c
@@ -384,7 +384,6 @@ static void exec_tick(struct exec *req)
 	PRE(req->leader != NULL && req->leader->db != NULL);
 	struct leader *leader = req->leader;
 	struct db *db = leader->db;
-	struct vfsTransaction transaction;
 
 	for (;;) {
 		leader_trace(leader, "exec tick %s (status = %d)",
@@ -515,37 +514,42 @@ static void exec_tick(struct exec *req)
 			if (req->status != RAFT_OK) {
 				sm_move(&req->sm, EXEC_DONE);
 				continue;
-			}
+			} else {
+				struct vfsTransaction transaction = {};
+				int rc = VfsPoll(leader->conn, &transaction);
+				if (rc != SQLITE_OK) {
+					leader_trace(leader,
+						     "poll failed on leader");
+					rc = VfsAbort(leader->conn);
+					assert(rc == SQLITE_OK);
+					req->status = RAFT_IOERR;
+					sm_move(&req->sm, EXEC_DONE);
+					continue;
+				}
 
-			int rc = VfsPoll(leader->conn, &transaction);
-			if (rc != SQLITE_OK) {
-				leader_trace(leader, "poll failed on leader");
-				rc = VfsAbort(leader->conn);
-				assert(rc == SQLITE_OK);
-				req->status = RAFT_IOERR;
-				sm_move(&req->sm, EXEC_DONE);
-				continue;
-			}
+				leader_trace(leader,
+					     "polled connection (%d frames)",
+					     transaction.n_pages);
+				if (transaction.n_pages == 0) {
+					sm_move(&req->sm, EXEC_DONE);
+					continue;
+				}
 
-			leader_trace(leader, "polled connection (%d frames)", transaction.n_pages);
-			if (transaction.n_pages == 0) {
-				sm_move(&req->sm, EXEC_DONE);
-				continue;
-			}
+				req->status = exec_apply(req, &transaction);
+				for (unsigned i = 0; i < transaction.n_pages;
+				     i++) {
+					sqlite3_free(transaction.pages[i]);
+				}
+				sqlite3_free(transaction.pages);
+				sqlite3_free(transaction.page_numbers);
 
-			req->status = exec_apply(req, &transaction);
-			for (unsigned i = 0; i < transaction.n_pages; i++) {
-				sqlite3_free(transaction.pages[i]);
+				if (req->status != 0) {
+					sm_move(&req->sm, EXEC_DONE);
+					continue;
+				}
+				sm_move(&req->sm, EXEC_WAITING_APPLY);
+				suspend;
 			}
-			sqlite3_free(transaction.pages);
-			sqlite3_free(transaction.page_numbers);
-
-			if (req->status != 0) {
-				sm_move(&req->sm, EXEC_DONE);
-				continue;
-			}
-			sm_move(&req->sm, EXEC_WAITING_APPLY);
-			suspend;
 		case EXEC_WAITING_APPLY:
 			sm_move(&req->sm, EXEC_DONE);
 			continue;

